commands/find_stemcell_version_test.go

```diff
@@ -0,0 +1,243 @@
+  version: "1.2.3"
+  remote_source:
+  remote_path: my-remote-path
+stemcell_criteria:
+  os: some-os
+  version: "4.5.6"
+`
+		})
+
+		JustBeforeEach(func() {
+
+			_, requestErr = pivnet.Do(simpleRequest)
+			Expect(requestErr).NotTo(HaveOccurred())
+
+			tmpDir, err := ioutil.TempDir("", "fetch-stemcell-test")
+
+			someKilnfilePath = filepath.Join(tmpDir, "Kilnfile")
+
+			err = ioutil.WriteFile(someKilnfilePath, []byte(kilnfileContents), 0644)
+			Expect(err).NotTo(HaveOccurred())
+
+			someKilnfileLockPath = filepath.Join(tmpDir, "Kilnfile.lock")
+			err = ioutil.WriteFile(someKilnfileLockPath, []byte(lockContents), 0644)
+			Expect(err).NotTo(HaveOccurred())
+
+			findStemcellVersion = commands.NewFindStemcellVersion(logger, &pivnet)
+
+			fetchExecuteArgs = []string{
+				"--kilnfile", someKilnfilePath,
+			}
+			executeErr = findStemcellVersion.Execute(fetchExecuteArgs)
+		})
+
+		When("stemcell criteria does not exist in the kilnfile", func() {
+			BeforeEach(func() {
+				kilnfileContents = `
+release_sources:
+- type: s3
+  bucket: compiled-releases
+  region: us-west-1
+  access_key_id: my-access-key-id
+  secret_access_key: my-secret-access-key
+  path_template: 2.8/{{trimSuffix .Name "-release"}}/{{.Name}}-{{.Version}}-{{.StemcellOS}}-{{.StemcellVersion}}.tgz
+  publishable: true
+`
+			})
+			It("returns the stemcell os info missing error message", func() {
+				Expect(executeErr).To(HaveOccurred())
+				Expect(executeErr).To(MatchError(ContainSubstring(commands.ErrStemcellOSInfoMustBeValid)))
+			})
+		})
+
+		When("stemcell major version does not exist in the kilnfile", func() {
+			BeforeEach(func() {
+				kilnfileContents = `
+release_sources:
+- type: s3
+  bucket: compiled-releases
+  region: us-west-1
+  access_key_id: my-access-key-id
+  secret_access_key: my-secret-access-key
+  path_template: 2.8/{{trimSuffix .Name "-release"}}/{{.Name}}-{{.Version}}-{{.StemcellOS}}-{{.StemcellVersion}}.tgz
+  publishable: true
+stemcell_criteria:
+  os: ubuntu-xenial
+`
+			})
+
+			It("returns stemcell major version missing error message", func() {
+				Expect(executeErr).To(HaveOccurred())
+				Expect(executeErr).To(MatchError(ContainSubstring(commands.ErrStemcellMajorVersionMustBeValid)))
+			})
+		})
+
+		When("stemcell OS and major version is specified", func() {
+			When("a new stemcell exists", func() {
+				BeforeEach(func() {
+					serverMock.Results.Res.Body = fakes.NewReadCloser(`{"version": "456.118"}`)
+					serverMock.Results.Res.StatusCode = http.StatusOK
+					serverMock.Results.Err = nil
+				})
+
+				It("returns the latest stemcell version", func() {
+					Expect(executeErr).NotTo(HaveOccurred())
+					Expect((&writer).String()).To(ContainSubstring("\"456.118\""))
+					Expect((&writer).String()).To(ContainSubstring("\"remote_path\":\"network.pivotal.io\""))
+					Expect((&writer).String()).To(ContainSubstring("\"source\":\"Tanzunet\""))
+				})
+			})
+		})
+	})
+
+	Describe("Execute", func() {
```

Comment:
This should probably be named "ExtractMajorVersion" the previous `Describe` is testing "Execute".

Reply:
Yhea this one was a typo 👍